src/forms/create_ticket.ts

```diff
@@ -35,9 +39,9 @@ export async function newCreateTicketForm(call: AppCall): Promise<AppForm> {
 class FormFields extends BaseFormFields {
     zdTicketForms: ZDFormFieldOption[]
     postMessage: string
-    constructor(call: AppCall, zdClient: ZDClient, mmClient: Client4) {
+    constructor(call: AppCallRequest, zdClient: ZDClient, mmClient: Client4) {
         super(call, zdClient, mmClient);
-        this.postMessage = call.context.post.message;
+        this.postMessage = call?.context?.post?.message as string;
```

Comment:
I'd think the first CallWithContext/CallWithPost part should work, but I can't think of if there's any way to shorthand that. I don't think there's a good generic way to tell TypeScript "yeah, that says optional, but I know it's set" since that sounds like what you're asking about though

Reply:
@mickmister, @hmhealey please see the latest changes to this file and the creation of the following file.

https://github.com/mattermost/mattermost-app-zendesk/blob/lint-fixes/src/types/apps.ts

Reply:
That works for me.

The other thing I could think to do for that would be to just have two objects passed around: the AppContext and some generic Context that could be modified depending on caller. That simplifies typing, but obviously makes it more annoying to work with since you're passing two objects around instead of once.

Reply:
To solve this problem, I created some concrete Ctx* types that are used when the request is read.  Any function calls that were using AppContext were changed to access a finite number of inputs. (mattermost_site_url, bot_access_token, .etc)